data/python.gram

```diff
@@ -1580,10 +1580,10 @@ invalid_arguments[NoReturn]:
             "invalid syntax. Maybe you meant '==' or ':=' instead of '='?", a, b
         )
      }
-    | a=args for_if_clauses {
-        self.store_syntax_error_starting_from(
+    | a=args b=for_if_clauses {
+        self.store_syntax_error_known_range(
             "Generator expression must be parenthesized",
-            a[1][-1] if a[1] else a[0][-1]
+            a[1][-1] if a[1] else a[0][-1], b
```

Comment:
Hummm, not sure this works, you need to get the node that is further away in the comprehension. Check the c code helper in the CPython repo

Reply:
Yeah I realized based on the failed checks, I need b[-1], I think I will test it before pushing this time.